goldenrun.py

```diff
@@ -418,3 +419,45 @@ def process_wildcard_faults(faultconfig, tbexec, tbinfo):
         new_fault_entry["faultlist"] = [wildcard_faults[i]]
         new_fault_entry["delete"] = False
         faultconfig.append(new_fault_entry)
+
+
+def process_single_faults(faultconfig, tbexec, tbinfo):
+    remove_list = []
+    for faultentry in faultconfig:
+        for fault in faultentry["faultlist"]:
+            if not isinstance(fault.mask, dict):
+                continue
+
+            tbinfo_tb_indexed = tbinfo.set_index("id")
+            for tb in tqdm(tbexec["tb"], leave=False):
+                tb_info_asm = tbinfo_tb_indexed.at[tb, "assembler"]
+                tb_info_total_size = tbinfo_tb_indexed.at[tb, "size"]
+
+                tb_info_asm = [
+                    int(instr.split("]")[0], 16)
+                    for instr in tb_info_asm.split("[ ")[1:]
+                ]
+                tb_info_size = list(numpy.diff(tb_info_asm))
+                tb_info_size.append(
+                    tb_info_total_size - sum(tb_info_size)
+                )  # calculate the last instr size
+
+                if fault.address not in tb_info_asm:
```

Comment:
Instead of doing this here, it might be more efficient to use `find_insn_addresses_in_tb` right at the start of the TB for loop (line 433).

Reply:
I checked it with a benchmark. `find_insn_addresses_in_tb` requires tbinfo in a bit different format. Including the time it takes to convert, using it provides neglectable improvement with more code. So instead of that, I moved the existing check earlier to save some computation.

Reply:
Thanks for checking!